README.md

```diff
@@ -79,15 +97,6 @@ cargo build --release
 | [weaver registry emit](docs/usage.md#registry-emit)                       | Emits a semantic convention registry as example signals to your OTLP receiver |
 | [weaver completion](docs/usage.md#completion)                             | Generate shell completions |
 
-## Documentation
```

Comment:
I realize you didn't aim for completeness, but until we can ensure we have links to all these things, I think we should at least keep this section somewhere below your new sections.

Generally like the improvements and flow, but would like to keep these links until we determine they're no longer needed.

Reply:
@jsuereth @tombrk 

I made an inventory of the links removed from the main README and the relocation of these links into the new documents introduced in this PR. Here’s the result:
- Links moved to `docs/codegen.md`:
  - `docs/architecture.md`
  - `docs/weaver-config.md`
  - `crates/weaver_forge/README.md`
- Links moved elsewhere in the main README:
  - `crates/weaver_live_check/README.md`
  - `docs/schema-changes.md`
  - application telemetry schema OTEP

The only link that I think has disappeared is `crates/weaver_checker/README.md`. It should be in `docs/validate.md` instead of the link to `crates/weaver_forge/README.md`, which was probably introduced by mistake.

I will add a comment in the `docs/validate.md` file to more explicitly point out this minor issue.